module 5.5

odianboy opened the pull request:
**ТЗ в файлах.**

odianboy committed:
module 5.5

NikitaKasovsky commented on module 5.2/3/tasks-module-3.js:
```diff
@@ -0,0 +1,29 @@
+        userNum = prompt('Введите Ваше число', 0);
+    } while ( !isFinite(userNum) );
+
+    if (userNum === null || userNum === '') {
```
такое условие можно сократить до `!userNum`

odianboy replied:
В userNum должна быть проверка на число, а prompt возвращает строку.

NikitaKasovsky commented on module 5.3/2/tasks-module-2.js:
```diff
@@ -0,0 +1,18 @@
+Функция должна быть нечувствительна к регистру.
+ */
+
+function checkSpam(str) {
```
лучше переделать на два аргумента, строка и массив исключаемых слов. Что бы постоянно не раздувать условия исключений

odianboy replied:
Реализовал. Не сделал через через forbiddenWords.includes(word), т.к он ищет прямое вхождение в массив (слово должно быть идентично тому, что в массиве), а по заданию надо чтобы было совпадение в слове. Т.е если делать через  forbiddenWords.includes(free xxxxx) == false , а по заданию оно должно быть true.

NikitaKasovsky commented on module 5.5/8/tasks-module-8.js:
```diff
@@ -0,0 +1,33 @@
+let petya = { name: "Петя", surname: "Иванов", id: 2 };
+let masha = { name: "Маша", surname: "Петрова", id: 3 };
+
+let users = [ vasya, petya, masha ];
```
Нагородил ты тут непонятно чего, эта задача решается так.

`users.map(user => {id: user.id, fullName: '${user.name} ${user.surname}'});`

odianboy replied:
Понял, исправил, спасибо

odianboy committed:
Update tasks-module-2.js

Не сделал через через forbiddenWords.includes(word), т.к он ищет прямое вхождение в массив (слово должно быть идентично тому, что в массиве), а по заданию надо чтобы было совпадение в слове. Т.е если делать через  forbiddenWords.includes(free xxxxx) == false , а по заданию оно должно быть true.

odianboy committed:
Update tasks-module-8.js
